lxd/api.go

```diff
@@ -157,9 +164,14 @@ func restServer(d *Daemon) *http.Server {
 		w.Header().Set("Content-Type", "application/json")
 
 		ua := r.Header.Get("User-Agent")
-		if uiEnabled && strings.Contains(ua, "Gecko") {
-			// Web browser handling.
-			http.Redirect(w, r, "/ui/", http.StatusMovedPermanently)
+		if strings.Contains(ua, "Gecko") {
+			if !uiEnabled {
+				w.Header().Set("Content-Type", "text/html")
+				w.WriteHeader(http.StatusNotImplemented)
```

Comment:
We should probably still return 200 OK so that any monitoring of the main listener's `/` endpoint doesn't start failing where it previously returned 200 OK, even if the UI is disabled, the service is still running.